src/libYARP_dev/include/yarp/dev/IRobotDescription.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2016 iCub Facility, Istituto Italiano di Tecnologia
+ * Authors: Marco Randazzo <marco.randazzo@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#ifndef YARP_DEV_IROBOTDESCRIPTION_H
+#define YARP_DEV_IROBOTDESCRIPTION_H
+
+#include <yarp/os/Vocab.h>
+#include <yarp/os/Log.h>
+#include <vector>
+#include <string>
+
+namespace yarp {
+    namespace dev {
+        class IRobotDescription;
+
+        struct RobotDescription
+        {
+            std::string  device_name;
+            std::string  device_type;
+            bool operator ==(RobotDescription const& b) { return this->device_name == b.device_name && this->device_type == b.device_type; }
+        };
+      }
+}
+
+/**
+ * @ingroup dev_iface_motor
+ *
+ * This interface allows users to retrieve a list which contains the names and the types of the currently running devices.
+ * This list is stored in a RobotDescriptionsServer class. The user can access to the data opening a RobotDescriptionsClient in its own module.
+ */
+class yarp::dev::IRobotDescription
+{
+public:
+    /**
+     * Destructor.
+     */
+    virtual ~IRobotDescription() {}
+
+    /**
+     * Ask the complete list of all yarp device drivers registered by a robot description server.
+     */
+    virtual bool getAllDevices(std::vector<RobotDescription>& dev_list) = 0;
+
+    /**
+    * Ask a list of all yarp control board wrappers (i.e. all the devices of type 'controlboardwrapper' and 'controlboardwrapper2')
+    * registered by a robot description server.
+    */
+    virtual bool getControlBoardWrapperDevices(std::vector<RobotDescription>& dev_list) = 0;
```

Comment:
I do not like this. YARP devices have a common base interface and are distinguished only by their name and interface they implement. This function is specific to a certain class of devices (ControlBoardWrapperDevices); if we add other devices or modify them we will need to change name of this function or add new ones. Can't this function be replaced by a generic: 

getAllDevicesByString(const std::string &key, std::vector<RobotDescription>& dev_list);

which return a list of device whose description or name matches the required keyword?